ding/framework/middleware/functional/logger.py

```diff
@@ -66,3 +74,84 @@ def _logger(ctx: "OfflineRLContext"):
                     writer.add_scalar('basic/train_{}-train_iter'.format(k), v, ctx.train_iter)
 
     return _logger
+
+
+def wandb_logger(cfg: EasyDict, env: BaseEnvManagerV2, model) -> Callable:
+    color_list = ["orange", "red", "blue", "purple", "green", "darkcyan"]
+    metric_list = ["q_value", "target q_value", "loss", "lr", "entropy"]
+    env.enable_save_replay(replay_path=cfg.record_path)
+    if cfg.gradient_logger:
+        wandb.watch(model)
+
+    def _action_prob(num, action_prob, ln):
+        ax = plt.gca()
+        ax.set_ylim([0, 1])
+        for rect, x in zip(ln, action_prob[num]):
+            rect.set_height(x)
+        return ln
+
+    def _value_prob(num, action_prob, action, ln):
+        ax = plt.gca()
+        ax.set_ylim([0, 1])
+        for rect, x in zip(ln, action_prob[num][action[num][0] - 1]):
+            rect.set_height(x)
+        return ln
+
+    def _plot(ctx: "OnlineRLContext"):
+        if not cfg.plot_logger:
+            return
+        for metric in metric_list:
+            if metric in ctx.train_output[0]:
+                metric_value = np.mean([item[metric] for item in ctx.train_output])
+                wandb.log({metric: metric_value})
+        if ctx.eval_value != -np.inf:
+            wandb.log({"reward": ctx.eval_value})
+
+        if ctx.eval_value != -np.inf:
+            if 'logit' in ctx.eval_output[0]:
+                action_value = [to_ndarray(F.softmax(v['logit'], dim=-1)) for v in ctx.eval_output]
+            if 'distribution' in ctx.eval_output[0]:
+                value_dist = [to_ndarray(v['distribution']) for v in ctx.eval_output]
+                action = [to_ndarray(v['action']) for v in ctx.eval_output]
+
+            file_list = []
+            for p in os.listdir(cfg.record_path):
+                if os.path.splitext(p)[-1] == ".mp4":
+                    file_list.append(p)
+            file_list.sort(key=lambda fn: os.path.getmtime(os.path.join(cfg.record_path, fn)))
+
+            fig, ax = plt.subplots()
+            plt.ylim([-1, 1])
+            if cfg.action_logger == 'q_value' or 'action probability':
+                action_dim = len(action_value[0])
+                x_range = [str(x + 1) for x in range(action_dim)]
+                ln = ax.bar(x_range, [0 for x in range(action_dim)], color=color_list[:action_dim])
+                ani = animation.FuncAnimation(
+                    fig, _action_prob, fargs=(action_value, ln), blit=True, save_count=len(action_value)
+                )
+                ani.save(cfg.record_path + "/" + str(ctx.env_step) + ".gif", writer='pillow')
+                wandb.log(
+                    {
+                        "video": wandb.Video(cfg.record_path + "/" + file_list[-2], format="mp4"),
+                        "q value": wandb.Video(cfg.record_path + "/" + str(ctx.env_step) + ".gif", format="gif")
+                    }
+                )
+            elif cfg.action_logger == 'q_value distribution':
```

Comment:
add some assertion for the dimension of variables like `value_dist`, which is beneficial to debug